freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -275,9 +328,37 @@ private SecretResponse getSecretResponseForKeytab(String accountId, String keyta
         }
     }
 
-    private String constructVaultPath(String accountId, String type, String subtype) {
-        return String.format("%s/%s/%s/%s-%s", accountId, type, subtype,
-                UUID.randomUUID().toString(), Long.toHexString(System.currentTimeMillis()));
+    private String constructVaultPathPrefix(CleanupRequest request, String accountId, String subType) {
```

Comment:
In overall this vault path construction got out of hand. It's very hard to understand what and why happens and how is it gonna look like. This really needs to be refactored or even moved out to sth like VaultPathBuilder. Also vault paths are constructed in these methods but still modified in the caller which makes it really hard to overview.

Reply:
There are type of API's 
1. constructVaultPath
2. constructVaultPathPrefix

Paths constructed by constructVaultPath are not updated by the caller.
Paths constructed by constructVaultPathPrefix are updated.
Let's there a host that is deleted, we need to identify all the vault paths associated with all the secrets for all the services in that host.

Reply:
I'm basically using the "list" API provided by the vault to identify the paths.
Let say a pathPerfix "/enginePath/appPath/account-id/Type/SubType/envCrn/clusterCrn/" is provided to the list API provided by the vault, it returns all the directories within that. 
For Example, If there were secrets stored for two hosts (10.32.140.201 and 10.32.140.202) in a given cluster, the API will return 10.32.140.201 and 10.32.140.202. We then have to get the location for the services with each host.

Reply:
I worked with Kalyan and posted fix for this.